src/ClientData/TimerDataTest.cpp

```diff
@@ -119,6 +119,22 @@ std::unique_ptr<TimerData> GetOrderedTimersSameDepth() {
   return timer_data;
 }
 
+std::unique_ptr<TimerData> GetUnorderedTimersSameDepth() {
```

Comment:
Just wondering. Do we really need to allocate TimerData on heap? Can't we use value semantics here?

Reply:
Can't we make the TimerData instances returned by `GetUnorderedTimersSameDepth` and `GetTimersDifferentDepths` const or constexpr and get rid of the two functions?

Reply:
> Just wondering. Do we really need to allocate TimerData on heap? Can't we use value semantics here?

I did that because TimerData has deleted its copy-constructor. 

> Can't we make the TimerData instances returned by GetUnorderedTimersSameDepth and GetTimersDifferentDepths const or constexpr and get rid of the two functions?

I also don't like the solution but I tried a few things quickly and I wasn't able to improve the code. Could you clarify a little bit what are you suggesting? We can briefly talk offline is this is preferred for you.

Reply:
I would disregard this comment for now. Things are a bit harder because absl::mutex is not movable. Feel free to ping me and I can address it later.